scilpy/reconst/multi_processes.py

```diff
@@ -0,0 +1,420 @@
+    B, _ = sh_to_sf_matrix(sphere, sh_order, sh_basis_type)
+
+    data_shape = shm_coeff.shape
+    if mask is not None:
+        shm_coeff = applymask(shm_coeff, mask)
+        peak_dirs = applymask(peak_dirs, mask)
+        peak_values = applymask(peak_values, mask)
+        peak_indices = applymask(peak_indices, mask)
+
+    nbr_processes = multiprocessing.cpu_count() if nbr_processes is None \
+        or nbr_processes < 0 else nbr_processes
+
+    npeaks = peak_values.shape[3]
+    shm_coeff = shm_coeff.ravel().reshape(np.prod(data_shape[0:3]), data_shape[3])
+    peak_dirs = peak_dirs.ravel().reshape((np.prod(data_shape[0:3]), npeaks, 3))
+    peak_values = peak_values.ravel().reshape(np.prod(data_shape[0:3]), npeaks)
+    peak_indices = peak_indices.ravel().reshape(np.prod(data_shape[0:3]), npeaks)
+    shm_coeff_chunks = np.array_split(shm_coeff, nbr_processes)
+    peak_dirs_chunks = np.array_split(peak_dirs, nbr_processes)
+    peak_values_chunks = np.array_split(peak_values, nbr_processes)
+    peak_indices_chunks = np.array_split(peak_indices, nbr_processes)
+    chunk_len = np.cumsum([0] + [len(c) for c in shm_coeff_chunks])
+
+    pool = multiprocessing.Pool(nbr_processes)
+    results = pool.map(maps_from_sh_parallel,
+                       zip(shm_coeff_chunks,
+                           peak_dirs_chunks,
+                           peak_values_chunks,
+                           peak_indices_chunks,
+                           itertools.repeat(B),
+                           itertools.repeat(sphere),
+                           itertools.repeat(gfa_thr),
+                           np.arange(len(shm_coeff_chunks))))
+    pool.close()
+    pool.join()
+
+    nufo_map_array = np.zeros((np.prod(data_shape[0:3])))
+    afd_map_array = np.zeros((np.prod(data_shape[0:3])))
+    afd_sum_array = np.zeros((np.prod(data_shape[0:3])))
+    rgb_map_array = np.zeros((np.prod(data_shape[0:3]), 3))
+    gfa_map_array = np.zeros((np.prod(data_shape[0:3])))
+    qa_map_array = np.zeros((np.prod(data_shape[0:3]), npeaks))
+    for i, nufo_map, afd_map, afd_sum, rgb_map, gfa_map, qa_map in results:
+        nufo_map_array[chunk_len[i]:chunk_len[i+1]] = nufo_map
+        afd_map_array[chunk_len[i]:chunk_len[i+1]] = afd_map
+        afd_sum_array[chunk_len[i]:chunk_len[i+1]] = afd_sum
+        rgb_map_array[chunk_len[i]:chunk_len[i+1], :] = rgb_map
+        gfa_map_array[chunk_len[i]:chunk_len[i+1]] = gfa_map
+        qa_map_array[chunk_len[i]:chunk_len[i+1], :] = qa_map
+
+    nufo_map_array = nufo_map_array.reshape(data_shape[0:3])
+    afd_map_array = afd_map_array.reshape(data_shape[0:3])
+    afd_sum_array = afd_sum_array.reshape(data_shape[0:3])
+    rgb_map_array = rgb_map_array.reshape(data_shape[0:3] + (3,))
+    gfa_map_array = gfa_map_array.reshape(data_shape[0:3])
+    qa_map_array = qa_map_array.reshape(data_shape[0:3] + (npeaks,))
+
+    return(nufo_map_array, afd_map_array, afd_sum_array,
```

Comment:
In case this grow, maybe using a dictionary for returning them would be cool.
like simply storing them like:
`output_dict['nufo_map_array'] = nufo_map_array`

and then you return the dictionary

Reply:
@frheault should I do it?